### PR TITLE
Moon Items

### PR DESCRIPTION
**Pull request by ZoeyZolotova**
Added the four moon trial heart pieces and the Fierce Deity's Mask to the item pool.

### CODE REVIEW

File: Items.cs [hunk 1]
```diff
@@ -890,10 +902,20 @@ public class Items
                 "Map: Romani Ranch",
                 "Map: Great Bay",
                 "Map: Stone Tower",
-                "Goron Racetrack Grotto"
+                "Goron Racetrack Grotto",
+                "One Mask",
```
Comment (Kjelli) [inline Items.cs]: What do these mean? "One Mask", "Two Masks" etc.

Comment (mzxrules) [reply Items.cs]: on the moon, the requirements for entering the sub dungeons are to give up 1, 2, 3, and 4 masks at a time

File: Build.cs [hunk 2]
```diff
@@ -289,15 +289,9 @@ private void WriteItems()
...
-                    itemId -= Values.NumberOfAreasAndOther;
-                }
-
-                if (ItemUtils.IsItemDefinedPastAreas(replacesItemId)) {
-                    // Subtract amount of entries describing areas and other
-                    replacesItemId -= Values.NumberOfAreasAndOther;
-                }
+                itemId -= ItemUtils.GetItemOffset(itemId);
+
+                replacesItemId -= ItemUtils.GetItemOffset(replacesItemId);
```
Comment (Kjelli) [inline Build.cs]: Not really that related to your changes, but is there any good reason why we mix items and areas? Is it for the dependency/conditional checks?

File: LogicMigrator/Migrator.cs [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+        private static void AddItemNames(List<string> lines)
+        {
+            if (lines[1] == "- Deku Mask")
+            {
+                lines[0] = "-version 1";
+                return;
+            }
+            lines.RemoveAll(line => line.StartsWith("-"));
+            var itemNames = new string[] {"Deku Mask", "Hero's Bow", "Fire Arrow", "Ice Arrow", "Light Arrow", "Bomb Bag (20)", "Magic Bean",
```
Comment (Kjelli) [inline LogicMigrator/Migrator.cs]: Isn't this already defined in another location?

Comment (ZoeyZolotova) [reply LogicMigrator/Migrator.cs]: It is, but there needs to be a copy here in order for the logic migration to work regardless of the codebase. I.e. if we rearrange the items list, this list will need to remain as it is.

File: Build.cs (hunk 2 above)
Comment (ZoeyZolotova) [reply Build.cs]: Yep exactly for that reason. Might have to rework it one day, but not now.

**Pull request merged**